rbac/management/principal/it_service.py

```diff
@@ -275,7 +280,36 @@ def get_service_accounts_group(self, group: Group, bearer_token: str, options: d
             service_account_principals=sap_dict, it_service_accounts=it_service_accounts, options=options
         )
 
-        return service_accounts
+        # If either the description or name filters were specified, we need to only keep the service accounts that
+        # match that criteria.
+        sa_description_filter = options.get(SERVICE_ACCOUNT_DESCRIPTION_KEY)
+        sa_name_filter = options.get(SERVICE_ACCOUNT_NAME_KEY)
+
+        if sa_description_filter or sa_name_filter:
+            filtered_service_accounts: list[dict] = []
+
+            for sa in service_accounts:
+                # Initialize both matches as "True", because having them at "False" would force us to make them "True"
+                # whenever one of the filters is not specified, since otherwise not specifying a filter would stop the
+                # service account to be included in the resulting list.
+                sa_description_filter_matches: bool = True
+                sa_name_filter_matches: bool = True
```

Comment:
I can see that you use the type annotations here but the rbac project doesnt use them otherwise, so it is maybe question to whole team ... in any case, the annotations aren't checked at the moment in this repo

Reply:
The reason why I use type annotations is because I'll get nice suggestions from VSCode while developing, and I figured that the rest of the team would also like having those suggestions too. However in this case it might be a bit unnecessary since the type can be easily inferred? What do you think?

Reply:
I dont say that I want to remove it, to be honest I dont know ... because it it not only one occurrence of this so removing doesnt solve it ... these 2 annotations are correct but there are annotations added by previous PRs that are not ... so what I am saying is that in python if we start to use annotations, we have to check them by some tool and we are not doing that yet ... from this PR perspective I believe we can let it as it is, this is probably topic for whole team discussion

Reply:
Oops, sorry! I did not want to stir things up :joy: I thought it'd be a helpful thing, but please let me know what's the outcome of that discussion —or if I could be there too to listen to opinions that'd be great!— so that I can stop/keep type hinting things :smile: :+1:

Reply:
that's fine, it just causes type annotation checker like mypy finds some output, it has no other effect